# This is a BitKeeper generated diff -Nru style patch.
#
# ChangeSet
#   2005/01/10 17:02:11-02:00 marcelo@logos.cnet 
#   Alan Cox: xfs_attrmulti_by_handle lack of bounds checking, reported by Coverity (from 2.6.10-ac)
# 
# fs/xfs/linux-2.4/xfs_ioctl.c
#   2005/01/10 17:01:35-02:00 marcelo@logos.cnet +5 -0
#   Alan Cox: xfs_attrmulti_by_handle lack of bounds checking, reported by Coverity (from 2.6.10-ac)
# 
diff -Nru a/fs/xfs/linux-2.4/xfs_ioctl.c b/fs/xfs/linux-2.4/xfs_ioctl.c
--- a/fs/xfs/linux-2.4/xfs_ioctl.c	2005-01-13 08:45:27 -08:00
+++ b/fs/xfs/linux-2.4/xfs_ioctl.c	2005-01-13 08:45:27 -08:00
@@ -514,6 +514,11 @@
 	if (error)
 		return -error;
 
+	if(am_hreq.opcount > 1024) {
+		VN_RELE(vp);
+		return -XFS_ERROR(ENOMEM);
+	}
+
 	size = am_hreq.opcount * sizeof(attr_multiop_t);
 	ops = (xfs_attr_multiop_t *)kmalloc(size, GFP_KERNEL);
 	if (!ops) {
